src/canvas.jl

```diff
@@ -44,35 +44,35 @@ end
 
 # Implementation of the digital differential analyser (DDA)
 function lines!(c::Canvas, x1::Number, y1::Number, x2::Number, y2::Number, color::UserColorType)
-    if (x1 < origin_x(c) && x2 < origin_x(c)) ||
-        (x1 > origin_x(c) + width(c) && x2 > origin_x(c) + width(c))
-        return c
-    end
-    if (y1 < origin_y(c) && y2 < origin_y(c)) ||
-        (y1 > origin_y(c) + height(c) && y2 > origin_y(c) + height(c))
-        return c
-    end
-    toff = x1 - origin_x(c)
-    px1 = toff / width(c) * pixel_width(c)
-    toff = x2 - origin_x(c)
-    px2 = toff / width(c) * pixel_width(c)
-    toff = y1 - origin_y(c)
-    py1 = pixel_height(c) - toff / height(c) * pixel_height(c)
-    toff = y2 - origin_y(c)
-    py2 = pixel_height(c) - toff / height(c) * pixel_height(c)
-    dx = px2 - px1
-    dy = py2 - py1
-    nsteps = abs(dx) > abs(dy) ? abs(dx) : abs(dy)
-    inc_x = dx / nsteps
-    inc_y = dy / nsteps
-    cur_x = px1
-    cur_y = py1
-    fpw = Float64(pixel_width(c))
-    fph = Float64(pixel_height(c))
+    mx = origin_x(c)
+    Mx = origin_x(c) + width(c)
+    (x1 < mx && x2 < mx) || (x1 > Mx && x2 > Mx) && return c
+
+    my = origin_y(c)
+    My = origin_y(c) + height(c)
+    (y1 < my && y2 < my) || (y1 > My && y2 > My) && return c
+
+    x2p(x) = (x - mx) / width(c) * pixel_width(c)
+    y2p(y) = pixel_height(c) - (y - my) / height(c) * pixel_height(c)
+
+    Δx = x2p(x2) - (cur_x = x2p(x1))
+    Δy = y2p(y2) - (cur_y = y2p(y1))
+
+    nsteps = abs(Δx) > abs(Δy) ? abs(Δx) : abs(Δy)
+    nsteps = min(nsteps, typemax(Int32))  # hard limit
```

Comment:
@johnnychen94, is this a reasonable fix ? :thinking:

Reply:
LGTM but actually we could expect `nsteps` to be integers here since `for _ in 1:10.5` is equivalent to `for _ in 1:10`.

```suggestion
    nsteps = min(floor(Int, nsteps), typemax(Int32))  # hard limit
```

Reply:
I remember now why I didn't cast nsteps to `Int`, it fails the CI.